campaignion_webform_widget/js/widget.js

```diff
@@ -49,18 +49,20 @@ function gaLinkerHandler() {
   }
 }
 
-function registerListeners() {
-  if (window.addEventListener) {
-    window.addEventListener("DOMSubtreeModified", function(){messageParent();}, true);
-    document.addEventListener("DOMContentLoaded", gaLinkerHandler);
-  } else if (window.attachEvent) {
-    window.attachEvent("onDOMSubtreeModified", function(){messageParent();});
-  }
+if (window.addEventListener) {
+  window.addEventListener("DOMContentLoaded", function() {
+    gaLinkerHandler();
+    messageParent();
+    window.addEventListener("DOMSubtreeModified", messageParent, true);
+  });
 }
-window.onload = function() {
-  messageParent();
-  registerListeners();
+else {
+  window.onload = function() {
```

Comment:
I agree we can get rid of `window.attachEvent`, however, I don't see a reason to keep a *fallback* on `window.onload` instead? I rather think we need to *always* call `messageParent()` again on load, in case there were some images loaded or scripts that add content and the document size changed.

Reply:
I think this is probably what the `DOMSubtreeModified` listener is supposed to be doing... I'd like to be more certain about this, but documentation is sparse as it belongs to a deprecated API and I'm having issues testing it via the `campaignion_standalone` box. As for the `window.onload` fallback, I only included it to represent a path for clients that don't have `window.addEventListener` - i.e. IE < 9 - but I'd be comfortable about removing it.

Reply:
We don't support IE < 9 more.

The "DOMSubtreeModified" listener informs the parent that the iframe needs to be resized when the DOM is modified but not when an image (without a specified height) that was always in the DOM has finished loading… (just an example why load is more reliable regarding the actual document length, widgets don't usually load big images)

Reply:
Yes, I agree that the reflow on image loading could be an issue... thank you for pointing it out. As per our conversation earlier, I'd be quite happy not to go ahead with this change in the interest of not breaking other uses of the widget, especially as this code is up for replacement anyway.